File: Python/selectionSort.py
from findSmaller import find_smallest


def selection_sort(arr: list):
    sorted_arr = []
    for i in range(len(arr)):
        smallest_index = find_smallest(arr)
        smallest_item = arr.pop(smallest_index)
        sorted_arr.append(smallest_item)
    return sorted_arr


print(selection_sort([13, 3, 2, 4]))

File: Deno/TypeScript/selectionSort.ts
import findSmallest from "./findSmallest.ts";

export default function selectionSort(list: any[]) {
  const newList: any[] = [];
  let smallestIndex = 0;
  let smallestItem: any;
  const listLength = list.length;
  for (let index = 0; index < listLength; index++) {
    smallestIndex = findSmallest(list);
    smallestItem = list[smallestIndex];
    list.splice(smallestIndex, 1);
    newList.push(smallestItem);
  }
  return newList;
}

console.log(selectionSort([5, 3, 2, 10]));
